File: 125-valid-palindrome/125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String str) {
        
        str = str.toLowerCase();
        
        int l = 0;
        int h = str.length() - 1;
        
        while(l <= h){
            char cl = str.charAt(l);
            char ch = str.charAt(h);
            
            if(!(( cl >= 'a' && cl <= 'z')  || ( cl >= '0' && cl <= '9'))){
                l++;
            }

            else if(!(( ch >= 'a' && ch <= 'z')  ||  ( ch >= '0' && ch <= '9') )){
                h--;
            }
            
            else if(cl == ch){
                l++;
                h--;
            }
            
            else
               return false;     

        }
        
        return true;
    }
}

File: 125-valid-palindrome/125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    
   let arr = []
   
   for(let i=0; i<s.length; i++){
       
       if((s[i] >= "A" && s[i]  <= "Z") || (s[i] >= "a" && s[i] <= "z") || (s[i] >= "0" && s[i] <= "9")){
           arr.push(s[i].toLowerCase())
       } 
   }
    
    let seedha = arr.join("");
    let ulta = arr.reverse().join("")
    console.log(seedha, ulta)
    return seedha == ulta;
    
};